### PR TITLE
Rustdoc nocapture

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Following https://github.com/rust-lang/rust/pull/86230.

It might need to wait for rustdoc latest nightly to be published first to pass CI though (so tomorrow).

Also: the `--nocapture` option is unstable for now in rustdoc, is there something particular to be done until it's stable on cargo side? For now I simply add `-Zunstable-options` but not sure if it's the way to go?

### CODE REVIEW

File: src/cargo/ops/cargo_test.rs [hunk 1]
```diff
@@ -233,7 +233,15 @@ fn run_doc_tests(
         }
 
         for arg in test_args {
-            p.arg("--test-args").arg(arg);
+            // We special-case "--nocapture" because rustdoc needs to handle it directly.
+            if *arg == "--nocapture" {
+                // FIXME(GuillaumeGomez): remove the `unstable-options` once rustdoc stabilizes the
+                // `--notcapture` option.
```
Comment (tristan957) [inline src/cargo/ops/cargo_test.rs:239]: nocapture instead of notcapture

Comment (GuillaumeGomez) [reply src/cargo/ops/cargo_test.rs]: Thanks!

**Comment by ehuss** [comment]
Closing as this is a bit stale. I'm also not clear what this was needed, since it looks like one can just pass `-Zunstable-options` themselves.  Before reopening, please work with the Cargo team to settle on a design for whatever is needed here.

**Comment by MingweiSamuel** [comment]
The current cargo doesn't work, I assume because `--nocapture` gets prefixed with `--test-args`. Right now `cargo test -- --nocapture` still hides input from doctests because it doesn't forward the `--nocapture` argument properly. This fixes that. See the referenced issues.

**Comment by veryjos** [comment]
Is this fixed?

**Comment by ptrself** [comment]
Doesn't seem like.. but in the interim, try this:

```
RUSTDOCFLAGS="-Z unstable-options --nocapture" cargo +nightly test --doc
```

**Comment by metasim** [comment]
@ehuss @GuillaumeGomez Can you please consider reopening this? It seems the hard work has already been done in rustdoc, and but given most people use rustdoc via cargo, it's availability is quite hindered.

**Comment by GuillaumeGomez** [comment]
I can't reopen it so I'll let it up to cargo team.

**Pull request closed without merging**